test/e2e/custom-networking/custom_networking_test.go

```diff
@@ -0,0 +1,130 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package custom_networking
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/manifest"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/apps/v1"
+	coreV1 "k8s.io/api/core/v1"
+)
+
+var _ = Describe("Custom Networking Test", func() {
+
+	var (
+		deployment *v1.Deployment
+		podList coreV1.PodList
+		podLabelKey string
+		podLabelVal string
+		port int
+		replicaCount int
+		shouldConnect bool
+	)
+
+	Context("when creating deployment targeted using ENIConfig", func() {
```

Comment:
Would be good to add a negative test case as well. We could consider verifying the code flow when there is no ENIconfig assigned to a node..

Reply:
Yes, Will add!

Reply:
Added negative workflow test!